### ISSUE TITLE
Out of Sync Items list unexpectedly empty

### ISSUE THREAD

**Issue by squidpickles** [syncthing/syncthing#8018]
### Does your log mention database corruption?
No

### Include required information
 - Syncthing  v1.18.2, macOS (64-bit Intel/AMD)  Build 1979-12-31 (noupgrade) 
 - Firefox 93.0 (64-bit)
 
In the Remote Devices list, the other device shows `Out of Sync Items   45 items, ~336 MiB`. 

<img width="587" alt="Out of Sync Items   45 items, ~336 MiB" src="https://user-images.githubusercontent.com/962628/138504780-b91ad1da-69ec-4a31-aac7-3fb6fa4ed747.png">

But when I click on the link, a modal pops up and does not list any of these items.
<img width="941" alt="(empty modal)" src="https://user-images.githubusercontent.com/962628/138504830-c2962cae-e326-43d1-9b13-0662fcc4ca2c.png">

On the other device, the GUI shows `Up to Date` for this local device.

While I imagine this sync state comes from slightly different ignore rules or something, I can't easily troubleshoot without seeing the list of items.

**Comment by ChristianKuehnel** [comment]
I can confirm the issue on 2 Linux installations running v1.18.5. One of my folders is working fine, the other one has that error. The sync with my Windows machines is working fine for both folders. 

The issue happens with Firefox and Safari.

The files seem to be synced successfully, I just did a sporadic check of the SHA hashes.

Deleting the folder on the affected machine and re-configuring it does not solve the problem.

**Comment by lpv-repo** [comment]
Same here. I have 3 folders synced with android and windows, windows says there's files out of sync on my phone but they're actually synced, and the list is empty. All 3 folders are shown 'Up to Date'.

**Comment by dovy6** [comment]
I have the same problem. It seems like a file is confusing syncthing. I tried un-sharing and re-sharing the affected folder and when un-shared it shows 100 percent synced but when re-shared it goes back to showing files out of sync. The files do appear to be syncing fine, though. 

![image](https://user-images.githubusercontent.com/76541192/147245811-cc429116-4f4d-4e25-81df-684ed6fc613c.png)

Please let me know if there's anything further I can share to help resolve this issue. 
Thanks

**Comment by turbozapekanka** [comment]
The same issue for me, running 1.18.5 on Synology NAS and 1.18.3 on Android phone:

![image](https://user-images.githubusercontent.com/20775323/147782420-20bb03e1-213f-4f43-909e-d4c5ac07455b.png)

![image](https://user-images.githubusercontent.com/20775323/147782460-8e3adaab-3039-414b-a490-5bea9de37162.png)

.stignore file:
.*
@*

**Comment by QingXuDw** [comment]
Same issue for me. 1.18.6 on windows10 both side.
![image](https://user-images.githubusercontent.com/49812095/149660142-96679bce-d134-443a-a024-40305c5564a9.png)
There is only one file: test.txt in this folder.

**Comment by sgleizes** [comment]
For me, resetting the syncthing database restored everything to normal. Of course this is a workaround and does not "solve" the main issue here.

I had the issue on my phone where I was able to reset the database from the app directly.
For desktops, see https://docs.syncthing.net/users/syncthing.html#cmdoption-reset-database.

**Comment by tayanov** [comment]
+1

**Comment by WonderfulSoap** [comment]
same problem, I even don't know what happened

**Comment by ilvez** [comment]
I have same issue with 1.19.1. I'm sharing one folder between Linux and Windows machines. Directory was in sync and everything was fine. I wanted to move directory in Windows machine.

1. I stopped/removed folder from Syncthing in Windows.
2. Moved the folder to new place.
3. Shared again from Linux.
4. On Windows side I just picked the location of new folder.
5. After adding the folder, it was in status "Stopped". Folder was missing `.stfolder`.
6. I created .stfolder with Cygwin: `mkdir .stfolder`
7. After Re-scan folder shows "Up to date", but Remote Devices Linux is in "Syncing XX %, YY GB".

I'm trying to `--reset-database` to cure it. Will update this comment, if any change.

**Comment by Littlemac123** [comment]
i have this problem two! on linux ubuntu 20.4 64bit

**Comment by rptb1** [comment]
I had this problem recently, and posted https://forum.syncthing.net/t/understanding-items-and-sizes/19584 about it, ,my theories, and how I worked around it.  I'm linking it from here in case that's useful.  If I figure out a repro I'll post it here.

**Comment by mb720** [comment]
On Arch Linux, this seems to have fixed the issue:

  * `systemctl --user stop syncthing.service`
  * `syncthing --reset-database`
  * `systemctl --user start syncthing.service`

**Comment by tomasz1986** [comment]
Yeah, resetting the database, which has already been mentioned at least a few times above, does get rid of the empty items, but at the same time, it doesn't help to determine the actual culprit.

Database reset is also not always safe, e.g. if devices aren't 100% in sync or when using advanced options like `ignoreDelete`, so you should be careful when performing it.

**Comment by mb720** [comment]
Right, I'm interested in a more permanent solution as well since this kind of issue happened a few times over the last years. After resetting the Syncthing database on a Linode server (folder types are all "Receive Encrypted"), my Raspberry Pi has retransmitted all the data over the last two days.

**Comment by imsodin** [comment]
Needs triage label removal doesn't mean that there's no more info needed, just that whoever removed it believes this is an actual bug and deserves to stay open :)

The local/global state actual looks good, as the item numbers match. The slightly larger size is most likely due to the encryption - as in a larger size is expected, I don't know by how much. So what's left is the lower part of your table, which I presume indicates remote out of sync items?

You don't need to reset the entire DB, there are two less destructive options to try first:

1. Reset deltas to exchange all metadata again, by starting syncthing with --reset-deltas. If the issue reappers, that might present an opportunity for debugging: Do the same thing again, and with db,model debug logging enabled and capture the logs.
2. Only needed if 1. doesn't help: Just remove the one folder in the web UI on Outpost, delete the data on disk (as it's encrypted can't be re-used) and re-add it. That's basically like resetting the DB, just for that one folder only.

**Comment by mh166** [comment]
Thanks for the reply and the useful information. You were indeed correct in assuming the table shows remote out of sync items. 👍 

I will gladly try out the two steps you outlined and will enable debug logging, if the opportunity arises. May I ask on which machine I shall issue this command? Since Step 2. is referring to Outpost, I would guess I should issue the `--reset-deltas` command there. However, I'd like to confirm with you first to have the best possible chance for debugging this.

**Comment by imsodin** [comment]
Given all devices show since out of sync items, or might work everywhere. However yes, outpost is likely the best since it all seems to be around that one.

**Comment by zero-thermo** [comment]
How do I reset database in docker? I'm running into this issue with the Syncthing Linuxserver.io v1.29.2 image.    

When I enter the container and run `syncthing --reset-database`, I get `INFO: Successfully reset database - it will be rebuilt after next start.`  

However, restarting the container still shows the same issue.

**Comment by LinuxOnTheDesktop** [comment]
Two points.

**1)** This thread leaves it unclear _how_ to reset the database (on desktops and laptops, and without containerisation). For, the manual to which [sgleizes](https://github.com/sgleizes) linked seems not to contain the relevant command. However, the command does exist and does work (at least on Linux, which is all that I tested). So:

```
$ syncthing --reset-database
[. .] INFO: Successfully reset database - it will be rebuilt after next start
```
and then one starts/restarts Syncthing.

**2)**  Can nothing be done at least about the very confusing empty 'out of sync' GUI boxes? To see a blank window, or a window containing only headings and nothing filed under the headings, frustrates, especially given that one is trying to troubleshoot a problem.

**Comment by calmh** [comment]
(leveldb)

**Comment by LinuxOnTheDesktop** [comment]
> (leveldb)

Eh?

**Comment by sofia-nep** [comment]
I would also like more clarification wrt the reason for closing.

**Comment by tomasz1986** [comment]
> I would also like more clarification wrt the reason for closing.

The problem was related to the old LevelDB database that was used in Syncthing v1.
